File: app/src/main/java/pers/nchz/thatmvpdemo/MainView.java
package pers.nchz.thatmvpdemo;

import android.os.Bundle;

import pers.nchz.thatmvp.presenter.ThatBasePresenter;
import pers.nchz.thatmvp.view.IThatBaseView;
import pers.nchz.thatmvp.view.ThatBaseView;

public class MainView extends ThatBaseView<MainPresenter> implements IMainView {
    @Override
    public void initView(Bundle savedInstanceState) {
        addSubView(new HeadView(),savedInstanceState,0);
    }

    @Override
    public int getRootLayoutId() {
        return R.layout.activity_main;
    }

    @Override
    public Class<? extends IThatBaseView> getInterface() {
        return IMainView.class;
    }

    @Override
    public void mainTest() {

    }
}

File: app/src/main/java/pers/nchz/thatmvpdemo/ui/view/MainView.kt
package pers.nchz.thatmvpdemo.ui.view

import android.os.Bundle
import com.nhcz500.base.base.view.BaseView
import pers.nchz.thatmvpdemo.R
import pers.nchz.thatmvpdemo.ui.presenter.MainPre

class MainView :BaseView<MainPre>() {
    override fun initView(savedInstanceState: Bundle?) {

    }

    override fun getRootLayoutId(): Int {
       return R.layout.activity_main
    }
}